docs/aws/audit/cloudformationmonitoring/rules/enable_stack_notifications.mdx

```diff
@@ -24,6 +24,250 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local system and configure it with your AWS account credentials. You can do this by running the following commands:
+
+   ```
+   pip install awscli
+   aws configure
+   ```
+   You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all CloudFormation stacks: Use the following command to list all the CloudFormation stacks in your AWS account:
+
+   ```
+   aws cloudformation list-stacks --query 'StackSummaries[*].[StackName]' --output text
+   ```
+   This command will return a list of all stack names.
+
+3. Check if notifications are enabled: For each stack, you can check if notifications are enabled by describing the stack and looking for the "NotificationARNs" attribute. Use the following command:
+
+   ```
+   aws cloudformation describe-stacks --stack-name <StackName> --query 'Stacks[*].[NotificationARNs]' --output text
+   ```
+   Replace `<StackName>` with the name of the stack you want to check. If the output is empty, it means that notifications are not enabled for this stack.
+
+4. Repeat step 3 for all stacks: You need to repeat the previous step for all the stacks listed in step 2. This way, you can check if notifications are enabled for all your CloudFormation stacks.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary modules and establish a session with AWS using your credentials. Here's a basic setup:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
+    region_name='us-west-2' # or any other region you prefer
+)
+```
+
+3. Now, create a CloudFormation client from your session:
+
+```python
+cf = session.client('cloudformation')
+```
+
+4. Use the `describe_stacks()` function to get a list of all stacks, then check if each stack has notifications enabled. If the 'NotificationARNs' field is empty, then notifications are not enabled for that stack:
+
+```python
+response = cf.describe_stacks()
+
+for stack in response['Stacks']:
+    if not stack['NotificationARNs']:
+        print(f"Stack {stack['StackName']} does not have notifications enabled.")
+```
+
+This script will print out the names of all stacks that do not have notifications enabled. You can modify it to suit your needs, for example by writing the names to a file or sending an email alert.
+</Accordion>
+
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>